derive_client/utils/w3.py

```diff
@@ -217,9 +224,10 @@ def build_standard_transaction(
 
 def wait_for_tx_receipt(w3: Web3, tx_hash: str, timeout=120, poll_interval=1) -> AttributeDict:
```

Comment:
I have not yet replaced the use of `wait_for_tx_receipt` with `wait_for_tx_finality`, because the ERC20 approve + transfer flows currently require two transactions. We might not want to wait that long between approval and transfer. Moreover, we risk succeeding the approval and failing the transfer. We can use multicall to achieve atomicity (t will also require less RPC calls, i.e. building and simulating tx)